File: problems/greedy/TICT3_1_minimum_number_of_coins/solution.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int minimumCoins(int amount) {
        int count = 0;
        std::vector<int> coinTypes = {500, 100, 50, 10};
        
        for (int coin: coinTypes) {
            count += amount / coin;
            amount %= coin;
        }
        
        return count;
    }
};

int main() {
    int amount = 1260;
    Solution s = Solution();
    std::cout << s.minimumCoins(amount) << std::endl;
}

File: problems/greedy/TICT3_1_minimum_number_of_coins/solution.py
class Solution:
  def minimumCoins(self, amount: int) -> int:
    count = 0
    coin_types = [500,100,50,10]

    for coin in coin_types:
      print(amount)
      count += amount // coin
      print(count)
      amount %= coin
    
    return count

def main():
  amount = 1260
  s = Solution()
  print(s.minimumCoins(amount))

main()